test/Test/Data/Mutable/Array.hs

```diff
@@ -1,16 +1,112 @@
+{-# LANGUAGE LinearTypes #-}
 {-# LANGUAGE OverloadedStrings #-}
-{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
 
-module Test.Data.Mutable.Array (mutArrTests) where
+-- |
+-- Tests for mutable arrays.
+--
+-- TODO:
+--  * Test failures for out of bound access
+module Test.Data.Mutable.Array
+  ( mutArrTests,
+  )
+where
 
+import qualified Data.Array.Mutable.Linear as Arr
+import Data.Unrestricted.Linear
 import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import qualified Prelude.Linear as Lin
+import Test.Tasty
+import Test.Tasty.Hedgehog (testProperty)
+import Test.Util
 
-mutArrTests :: IO Bool
-mutArrTests = checkParallel group
+-- # Exported Tests
+--------------------------------------------------------------------------------
 
-group :: Group
-group = Group "Mutable array tests" $
-  []
+mutArrTests :: TestTree
+mutArrTests = testGroup "Mutable array tests" group
 
+group :: [TestTree]
+group =
+  [ testProperty "[read i (write arr i x) = x]" testWrite,
```

Comment:
There are very few tests. We could test that `read i (write arr j x) = read arr i` when `(j <> i)` (there are challenges due to linearity, but we can easily generate the same array twice).

We could test that reading twice at the same index gives the same result.

Maybe we need a test relating `length` and `resize`.

Etc…

Reply:
Yes, I'll add a lot more tests and a thorough reasoned scheme to covering everything important.